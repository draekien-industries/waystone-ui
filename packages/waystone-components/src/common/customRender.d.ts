/** @jsxImportSource theme-ui */
import React from 'react';
import { RenderOptions } from '@testing-library/react';
declare const DarkModeWrapper: React.FC;
declare const themedRender: (ui: React.ReactElement, options?: Omit<RenderOptions<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>, "wrapper"> | undefined) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>;
export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
//# sourceMappingURL=customRender.d.ts.map