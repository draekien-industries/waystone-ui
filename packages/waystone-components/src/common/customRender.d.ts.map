{"version":3,"file":"customRender.d.ts","sourceRoot":"","sources":["customRender.tsx"],"names":[],"mappings":"AAAA,gCAAgC;AAChC,OAAO,KAAoB,MAAM,OAAO,CAAC;AACzC,OAAO,EAAU,aAAa,EAAE,MAAM,wBAAwB,CAAC;AAQ/D,QAAA,MAAM,eAAe,EAAE,KAAK,CAAC,EAQ5B,CAAC;AAEF,QAAA,MAAM,YAAY,OACZ,MAAM,YAAY,sQAE8B,CAAC;AAEvD,cAAc,wBAAwB,CAAC;AACvC,OAAO,EAAE,YAAY,IAAI,MAAM,EAAE,eAAe,EAAE,CAAC"}