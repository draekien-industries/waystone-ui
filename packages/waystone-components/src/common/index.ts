export * from './customRender';
export * from './interfaces';
