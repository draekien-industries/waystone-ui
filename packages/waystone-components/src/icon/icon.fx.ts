import { IconVariant } from './icon.types';

export const getIconClassName = (variant?: IconVariant, className?: string) => {
  const trimmed = variant?.trim() as IconVariant;

  return trimmed === 'filled'
    ? `material-icons ${className || ''}`.trimEnd()
    : `material-icons-${trimmed} ${className || ''}`.trimEnd();
};
