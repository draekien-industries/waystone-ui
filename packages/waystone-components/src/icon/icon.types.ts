import { HasColor, HasSize } from '../common/interfaces';

export type IconVariant =
  | 'filled'
  | 'outlined'
  | 'round'
  | 'two-tone'
  | 'sharp';

/** The props required to generate the correct icon styles. */
export interface IconStyleProps extends HasSize, HasColor {}
