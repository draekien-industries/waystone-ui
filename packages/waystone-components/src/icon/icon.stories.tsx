import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Icon, IconProps, IconOptions } from './icon';



export default {
  title: 'Components/Icon',
  component: Icon,
  args: {
    variant: 'filled',
    size: 'md',
    color: 'inherit',
  },
} as Meta;

const Template: Story<IconProps> = (args) => <Icon {...args} />;

export const Default = Template.bind({});

Default.args = {
  name: 'accessibility',
};
