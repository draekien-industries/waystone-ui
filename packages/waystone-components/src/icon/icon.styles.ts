import { Size } from '@waystone/core';
import { ThemeUIStyleObject } from 'theme-ui';
import { IconStyleProps } from './icon';

enum IconSize {
  sm = '1em',
  md = '1.5em',
  lg = '2em',
}

const convert = (size: Size): IconSize =>
  IconSize[Size[size] as keyof typeof IconSize];

export const iconCss = (props: IconStyleProps): ThemeUIStyleObject => {
  const css: ThemeUIStyleObject = {
    verticalAlign: 'middle',
    color: props.color || 'inherit',
    fontSize: 'inherit',
  };

  if (props.size) {
    css.fontSize = `${convert(props.size)} !important`;
  }

  return css;
};
