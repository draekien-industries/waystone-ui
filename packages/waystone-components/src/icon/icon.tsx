/** @jsxImportSource theme-ui */
import React from 'react';
import { getIconClassName } from './icon.fx';
import { iconCss } from './icon.styles';

import { HasColor, HasSize, HasVariant } from '../common/interfaces';

export type IconVariant =
  | 'filled'
  | 'outlined'
  | 'round'
  | 'two-tone'
  | 'sharp';

/** The props required to generate the correct icon styles. */
export interface IconStyleProps extends HasSize, HasColor {}

/**
 * The options that can be set when rendering the icon.
 * @extends IconStyleProps
 * @extends HasVariant
 */
export interface IconOptions extends IconStyleProps, HasVariant {
  /**
   * The variant of icon to render.
   * @override
   */
  variant?: IconVariant;
}

/** Props for the icon component. */
export interface IconProps
  extends Omit<React.HTMLAttributes<HTMLElement>, 'children' | 'color'>,
    HasVariant,
    HasSize,
    HasColor {
  /** The name of the material icon to render. */
  name: string;
  /** The material icon variant to use when rendering the icon. */
  variant?: IconVariant;
}

/**
 * Renders an icon component.
 * This component library relies on the material-icons font for rendering icons.
 * You can view the list of available icons by going to https://fonts.google.com/icons
 */
export const Icon = ({
  className,
  name,
  variant = 'filled',
  size = 'md',
  color = 'inherit',
  ...rest
}: IconProps) => {
  return (
    <i
      sx={iconCss({ size, color })}
      className={getIconClassName(variant, className)}
      {...rest}>
      {name}
    </i>
  );
};
