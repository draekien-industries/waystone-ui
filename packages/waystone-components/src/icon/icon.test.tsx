import React from 'react';
import { render } from '../common';
import { Icon } from './icon';

describe('The Icon component', () => {
  it('Matches snapshot', () => {
    const { container } = render(<Icon name="accessibility" />);

    expect(container).toMatchSnapshot();
  });
});
