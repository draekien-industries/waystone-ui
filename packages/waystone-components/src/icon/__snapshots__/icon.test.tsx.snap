// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Icon component Matches snapshot 1`] = `
<div>
  <i
    class="material-icons css-1h9809z-Icon"
  >
    accessibility
  </i>
</div>
`;
